### PR TITLE
LIIKUNTA-582, LIIKUNTA-585 | feat: add hierarchical menus & universal bar menu support to Navigation

### PR DESCRIPTION
**Pull request by karisal-anders**
### feat: add hierarchical menus & universal bar menu support to Navigation

Navigation:
 - add support for hierarchical menus
 - add support for universal bar menu
   - the universal bar menu is shown on the top of the header at least
	 when using a wide screen
 - remove deprecated variant property

queries:
 - add parentId to menu query to fetch hierarchy of menus

mocks:
 - make navigation mock menu hierarchical
   - this hierarchical menu can be seen in storybook's ArchiveSearchPage
 - add mock for universal bar menu

packages:
 - add crypto-browserify as development dependency to fix running
   storybook
   - crypto is being used by hds-react since HDS v3
 - set version to 1.0.0-alpha230

refs LIIKUNTA-582

### Related

[LIIKUNTA-582](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-582)
[LIIKUNTA-585](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-585)

## Testing

In [PR](https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/552) review environments (*if they are working*):
 - https://liikunta-pr552.dev.hel.ninja/
 - https://harrastukset-pr552.dev.hel.ninja/
 - https://tapahtumat-pr552.dev.hel.ninja/

Locally:
 - Check out this PR's branch
 - `yarn`
 - `yarn storybook`

## Screenshots

image

## Additional notes

Published as canary build [react-helsinki-headless-cms@1.0.0-alpha230-canary-aa71ea3](https://www.npmjs.com/package/react-helsinki-headless-cms/v/1.0.0-alpha230-canary-aa71ea3) to npm

[LIIKUNTA-582]: https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LIIKUNTA-585]: https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core/navigation/Navigation.tsx [hunk 1]
```diff
@@ -33,17 +40,15 @@ export type NavigationProps = {
...
 };
 
-const LOGO_ARIA_LABELS = {
+const CITY_OF_HELSINKI_TRANSLATIONS = {
   EN: 'City of Helsinki',
   FI: 'Helsingin kaupunki',
   SV: 'Helsingfors stad',
 } as const satisfies Record<LanguageCodeEnum, string>;
 
-const LOGO_LABELS = {
```
Comment (melniiv) [inline src/core/navigation/Navigation.tsx:46]: we can add those though hcrc config, in `copy` section we have translations

File: src/core/navigation/Navigation.tsx [hunk 2]
```diff
@@ -120,7 +174,24 @@ export function Navigation({
   const logoProps: LogoProps = {
     size: 'large',
     src: LOGO_SOURCES[currentLanguageCode],
-    alt: LOGO_LABELS[currentLanguageCode],
+    alt: HELSINKI_TRANSLATIONS[currentLanguageCode],
+  };
+
+  const localizedCityOfHelsinki =
+    CITY_OF_HELSINKI_TRANSLATIONS[currentLanguageCode];
```
Comment (melniiv) [inline src/core/navigation/Navigation.tsx:181]: when translations are added though the config we dont need this code anymore

**Review by melniiv — APPROVED** [review]
LGTM, if translations are fixed

File: src/core/navigation/Navigation.tsx (hunk 1 above)
Comment (karisal-anders) [reply src/core/navigation/Navigation.tsx:46]: Moved to Config['translations']['helsinkiLogo']

File: src/core/navigation/Navigation.tsx [hunk 3]
```diff
@@ -33,17 +40,15 @@ export type NavigationProps = {
     allLanguages: Language[],
   ) => string;
   getIsItemActive?: (menuItem: MenuItem) => boolean;
-  /** @deprecated Not used anymore i.e. does nothing after HDS 3 was taken into use. */
-  variant?: 'default' | 'inline';
 };
 
-const LOGO_ARIA_LABELS = {
+const CITY_OF_HELSINKI_TRANSLATIONS = {
```
[the remark this answers is not in the document]
Comment (karisal-anders) [reply src/core/navigation/Navigation.tsx]: Moved these to config:
 - LOGO_ARIA_LABELS -> config.translations.cityOfHelsinki
 - LOGO_LABELS -> config.translations.helsinki
 - LOGO_SOURCES -> config.translations.helsinkiLogo

File: src/core/navigation/Navigation.tsx (hunk 2 above)
Comment (karisal-anders) [reply src/core/navigation/Navigation.tsx]: Moved these to config:
 - LOGO_ARIA_LABELS -> config.translations.cityOfHelsinki
 - LOGO_LABELS -> config.translations.helsinki
 - LOGO_SOURCES -> config.translations.helsinkiLogo

**Pull request merged**